util/src/validate.ts

```diff
@@ -178,27 +179,44 @@ function checkVertices(ids: string[], nodeModulesPath: string): void {
 		process.stdout.write(`${outputMessage}\r`);
 		count++;
 
-		const validation: ValidatorResult = validateSchema(vertex, vertexSchema);
-		if (!validation.valid) {
-			let errorMessage: string | undefined;
-			vertices[key].invalidate();
+		let errorMessage: string;
+		if (vertex.label === undefined) {
+			errorMessage = `requires property "label"`;
+		} else if (!Object.values(LSIF.VertexLabels).includes(vertex.label)) {
+			errorMessage = `unknown label`;
+		} else {
+			try {
+				let className: string = '';
+				if (vertex.label === LSIF.VertexLabels.event) {
+					if (vertex.scope === undefined) {
+						errorMessage = `requires property "scope"`;
+					} else {
+						className = vertex.scope[0].toUpperCase() + vertex.scope.slice(1) + 'Event';
+					}
+				} else {
+					className = vertex.label[0].toUpperCase() + vertex.label.slice(1);
```

Comment:
What's the reasoning for performing operations on the label like this? Seems like it would probably add some time if you're processing every string vs. just returning it as is

Reply:
Unfortunately the JSON schema generator I'm using is case sensitive, so I need to capitalize the label before calling the function, otherwise it won't find the class.